test/contrib/emscripten/conftest.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+import asyncio
+import contextlib
+import mimetypes
+import os
+import textwrap
+from pathlib import Path
+from typing import Any, Generator
+from urllib.parse import urlsplit
+
+import pytest
+from tornado import web
+from tornado.httputil import HTTPServerRequest
+
+from dummyserver.handlers import Response, TestingApp
+from dummyserver.testcase import HTTPDummyProxyTestCase
+from dummyserver.tornadoserver import run_tornado_app, run_tornado_loop_in_thread
+
+
+@pytest.fixture(scope="module")
+def testserver_http(
+    request: pytest.FixtureRequest,
+) -> Generator[PyodideServerInfo, None, None]:
+    dist_dir = Path(os.getcwd(), request.config.getoption("--dist-dir"))
+    server = PyodideDummyServerTestCase
+    server.setup_class(str(dist_dir))
+    print(
+        f"Server:{server.http_host}:{server.http_port},https({server.https_port}) [{dist_dir}]"
+    )
+    yield server
+    print("Server teardown")
+    server.teardown_class()
+
+
+class ServerRunnerInfo:
+    def __init__(self, host: str, port: int, selenium: Any) -> None:
+        self.host = host
+        self.port = port
+        self.selenium = selenium
+
+    def run_webworker(self, code: str) -> Any:
+        if isinstance(code, str) and code.startswith("\n"):
+            # we have a multiline string, fix indentation
+            code = textwrap.dedent(code)
```

Comment:
Is `dedent` safe to call always?

Reply:
yes - it's needed because otherwise you have to write any code strings on the first column and it messes up the indentation of your program.